attributes/default.rb

```diff
@@ -87,6 +87,12 @@
 # See README.md for details
 default['audit']['profiles'] = []
 
+# 'default['audit']' and 'default['audit']['profiles']' capabilities are extended to support symbol assignment.
+# @TODO: Once we fully support hash of hashes, then we may remove these extended methods and revert back to using:
+# default['audit']['profiles'] = {}
+default['audit'].extend(Deprecations::Audit::HashProfile)
+default['audit']['profiles'].extend(Deprecations::Audit::ArrayProfile)
```

Comment:
There's a bunch of different edge cases here that I can think of around this implementation, but instead of tracking those down what I'd suggest is to abandon this node attribute and pick a new name.  I'm going to call it `default['audit']['profiles-ng']` which is poor and should not be used.  But I can't think of anything better right this second.

Then in the current major version of the cookbook `default['audit']['profiles']` should throw a deprecation warning if it is set to anything other than an empty array.  If `default['audit']['profiles-ng']` is set to a non-empty hash then it should be used instead (there should be no merging and both-set should throw a warning that the old array setting is not being used.

Then in the next major version  of the cookbook `default['audit']['profiles']`  should just throw and fail if its set to anything other than an empty array, and only `default['audit']['profiles-ng']` should be set.

Reply:
@lamont-granquist thanks for the suggestion to store audit profiles within new variable `default['audit']['profiles-ng']`. Also we need minor refactor here https://github.com/chef-cookbooks/audit#configure-node

Reply:
Hey @lamont-granquist,

I like this approach, this is going to be a much simple solution to implement such deprecations.
But about the `default['audit']['profiles']` which are `already set for a node on the server and are not provided by the user while new chef-client run`, could you please let me know if I'm missing something while covering this crucial case:
> (there should be no merging and both-set should throw a warning that the old array setting is not being used.

We should provide backward compatibility to these preloaded `default['audit']['profiles']`, and also **load(merge) them along with `default['audit']['profiles-ng']`** followed by deletion of `default['audit']['profiles']`. later release will throw errors and fail if `default['audit']['profiles']` is set.

Reply:
I'm going to update the docs to not mention the deprecated method.